File: src/types/order.ts
/**
 * Tipos do sistema Mi Piace
 * Refletem os contratos reais da API (api.vfandrade.com)
 */

// ========================================
// Pedidos
// ========================================

export type OrderStatus = 'novo' | 'preparando' | 'entrega' | 'finalizado';
export type PaymentStatus = 'pago' | 'pendente';

/** Item de pedido como vem da API */
export interface OrderItem {
  id: string;
  product_id: string;
  quantity: number;
  details?: string;
  complement_ids: string[];
  products: { name: string };
}

/** Pedido como vem da API */
export interface ApiOrder {
  id: string;
  customer_name: string;
  customer_phone?: string;
  total_price: number;
  rua: string;
  numero: string;
  bairro: string;
  complemento_endereco?: string | null;
  status: string;
  payment_status: string;
  created_at: string;
  order_items: OrderItem[];
}

/** Pedido normalizado para o frontend */
export interface Order {
  id: string;
  customerName: string;
  customerPhone?: string;
  totalPrice: number;
  address: {
    rua: string;
    numero: string;
    bairro: string;
    complemento?: string | null;
  };
  status: OrderStatus;
  paymentStatus: PaymentStatus;
  createdAt: Date;
  items: OrderItem[];
}

// ========================================
// Inventário (Produtos + Grupos + Complementos)
// ========================================

export interface ApiProduct {
  id: string;
  name: string;
  base_price: number;
  is_available: boolean;
}

export interface ApiGroup {
  id: string;
  name: string;
  min_choices: number;
  max_choices: number;
  is_required: boolean;
}

export interface ApiComplement {
  id: string;
  group_id: string;
  name: string;
  extra_price: number;
  is_available: boolean;
}

/** Resposta completa de GET /products/ */
export interface InventoryResponse {
  products: ApiProduct[];
  groups: ApiGroup[];
  complements: ApiComplement[];
}

// ========================================
// Dashboard (mock — sem endpoint de métricas)
// ========================================

export interface SalesData {
  date: string;
  total: number;
  orders: number;
}

export interface ProductSalesData {
  name: string;
  sales: number;
  revenue: number;
}

export interface HourlySalesData {
  hour: string;
  orders: number;
  revenue: number;
}

File: back/app/schemas/order.py
from pydantic import BaseModel
from typing import List, Optional

class AddressSchema(BaseModel):
    rua: Optional[str] = None
    numero: Optional[str] = None
    bairro: Optional[str] = None
    complemento: Optional[str] = None

class OrderItemSchema(BaseModel):
    product_id: str
    selected_complement_ids: List[str] = []
    quantity: int = 1
    details: Optional[str] = None

class OrderCreate(BaseModel):
    customer_name: str
    customer_phone: str
    total_price: float
    chat_id: Optional[str] = None
    endereco: Optional[AddressSchema] = None
    items: List[OrderItemSchema]
    payment_id: Optional[str] = None
    payment_link: Optional[str] = None
    payment_method: Optional[str] = "pix"

class OrderUpdateStatus(BaseModel):
    status: Optional[str] = None
    payment_status: Optional[str] = None

# Schema para o n8n salvar os dados do Pix após gerá-lo
class OrderUpdatePayment(BaseModel):
    payment_id: str
    payment_link: str
    payment_method: str = "pix"